### ISSUE TITLE
Session Recording / Replay tools don't work

### ISSUE THREAD

**Issue by davidstellini** [QwikDev/partytown#241]
**Describe the bug**
Session recording tools (Like mouseflow, hotjar, and yandex) record user sessions & behaviour, they allow you to generate heatmaps of site usage and identify & fix issues by seeing exactly what users clicked on to great detail.

These tools are a massively good candidate for Partytown because they are, as you can imagine - heavy.

I made a repro with Yandex.

**To Reproduce**
1. Open https://davste93.github.io/angular-partytown-yandex-repro/
2. Click on home
3. Select some text
4. Close the tab
5. If it works, you should see a new session under: https://metrica.yandex.com/stat/visor?period=today&filter=(ym%3As%3AisRobot%3D%3D%2527No%2527)&id=89979942 (However you need to create a yandex account, give me your username, and I can add you to the dashboard so you can see it).
6. By running the script under JS directly, it works, but using text/partytown does not.

**Reproduction link**
https://davste93.github.io/angular-partytown-yandex-repro/

**Comment by davidstellini** [comment]
Updated:

Scrapped angular and used plain HTML.

I also added a version with Hotjar here: https://davste93.github.io/angular-partytown-yandex-repro/hotjar-partytown.html
Hotjar may be easier to debug because at least it's giving errors. It says: Access to XMLHttpRequest at 'https://vars.hotjar.com/box-1ada912494ba7fc7aca15fcef1c2a7ae.html' from origin 'https://davste93.github.io'

**Comment by jonathan-dejong** [comment]
Can confirm that Hotjar does not seem to work. From what I can tell the issue seems to be that while the `static.hotjar.com` host that the script tries to access works and has a `Access-Control-Allow-Origin: *` header it then sets up subsequent calls to `vars.hotjar.com` which does not and fails. 

```
Access to XMLHttpRequest at 'https://vars.hotjar.com/box-5e66f98b4ee957db209dc6f63e3d59dd.html' from origin 'https://www.*******.io' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.
set @ partytown-ww-sw.js?v=0.7.1:1193
window.hjBootstrap @ hotjar-1266105.js?sv=6:5
270 @ hotjar-1266105.js?sv=6:5
n @ hotjar-1266105.js?sv=6:5
eval @ hotjar-1266105.js?sv=6:5
eval @ hotjar-1266105.js?sv=6:5
run @ partytown-ww-sw.js?v=0.7.1:689
(anonymous) @ partytown-ww-sw.js?v=0.7.1:1684
await in (anonymous) (async)
receiveMessageFromSandboxToWorker @ partytown-ww-sw.js?v=0.7.1:1701
16:33:46.152 
```

**Comment by harrytran998** [comment]
Yeah, confirm this issue too. We cannot use a reverse proxy to control the `https://vars.hotjar.com/box-5e66f98b4ee957db209dc6f63e3d59dd.html`. Bcs this is a website 🥲 We cannot access the same domain so CORS will bummp.

We have the same issue when embedding some scripts to contain a website. So how we can do now @adamdbradley 🙏

**Comment by davidstellini** [comment]
Where are you guys seeing the CORS errors? I don't see any in the repros in the console?

**Comment by gomespereira** [comment]
Same CORS problem here.

Anyone tried to install Hotjar with Google Tag Manager and then setup the GTM script with Partytown?

**Comment by CesarBenavides777** [comment]
> Same CORS problem here.
> 
> 
> 
> Anyone tried to install Hotjar with Google Tag Manager and then setup the GTM script with Partytown?

Yes but you have to set up a CORS proxy. You can dump the code from CORS anywhere into a cloud flare worker and and set up resolveUrl there.

**Comment by harrytran998** [comment]
Anyone can investigate this issue 🥹 - Seemming this is a hard problem!

**Comment by PeretEtienne** [comment]
Has anyone managed to get hotjar to work with Partytown yet? I'm also curious to know how you manage to see the errors that happen in the service workers

**Comment by tcetin** [comment]
Has anyone solved the issue?

**Comment by kkopp001** [comment]
Has anyone successfully integrated any session replay tool with partytown? Or is it just impossible?

**Comment by gioboa** [comment]
Two likely blockers for session replay tools were fixed recently: cross-origin iframes created by worker scripts now load natively (#749), and scripts the worker can't fetch fall back to the main thread (#753). So I'm closing this issue for now, feel free to re-open it if it's still an issue for you.
